imports/ui/lists/recentAgents.coffee

```diff
@@ -75,40 +64,97 @@ Template.recentAgents.onRendered ->
       return
     pageHeight = document.documentElement.scrollHeight
     clientHeight = document.documentElement.clientHeight
-    if pageHeight - (scrollPos + clientHeight) < scroller.options.distance
-      scroller.updateInitiated = true
-      scroller.options.callback ->
-        scroller.updateInitiated = false
+    # Ignore scrolling if page is not overflowing the browser view by much
+    if pageHeight - clientHeight < scroller.options.distance * 2
+      return
+    # Handle scrolling
+    if scrollPos > prevScrollPos # scroll down
+      lastScrollDirection = 1
+      if pageHeight - (scrollPos + clientHeight) < scroller.options.distance
+        if prevScrollPos > pageHeight - (scrollPos + clientHeight)
+          console.log "down"
+          scroller.updateInitiated = true
+          scroller.options.callback ->
+            scroller.updateInitiated = false
+    else # scroll up
+      lastScrollDirection = -1
+      if scrollPos < scroller.options.distance
+        if prevScrollPos > scroller.options.distance
+          console.log "up"
+          scroller.updateInitiated = true
+          scroller.options.callbackUp ->
+            scroller.updateInitiated = false
     prevScrollPos = scrollPos
-
   options = {
-    distance: 50
+    distance: 250
     callback: (done) =>
-      @$("button.load-more-articles").click()
+      unless @mutex
```

Comment:
I've had some issues with debouncing it, will try to use it again once I have more items (articles) to test the pagination with. Thanks for the note.